GitUI/UserControls/RevisionGrid/Graph/ThreadSafeGrowingList.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace GitUI.UserControls.RevisionGrid.Graph
+{
+    // This list can only grow. Because it only grows, reading is thread safe.
+    // Writing to this list is not thread safe. However, for the usecase
+    // where only one thread is adding items to the list, and multiple threads
+    // are reading, this can be used.
+    public class ThreadSafeGrowingList<T> : IReadOnlyList<T>
+    {
+        private T[] _internalArray;
+
+        public ThreadSafeGrowingList(int capacity = 2)
+        {
+            _internalArray = new T[capacity];
+        }
+
+        public void Add(T item)
```

Comment:
❗️ This method is not thread safe. It should be updated, or the type should be renamed to not imply a contract the type does not adhere to.

Reply:
Agreed.

Is the idea that the collection is safe for concurrent readers and a single writer?

If concurrent writes are required, it should be possible to make this thread safe lock free by using two 'count' fields and interlocked operations (I think).

Reply:
Its meant to be used with only a single writer. Otherwise the existent Concurrenct* classes are available. I will make some changes and push to this PR, thanks for the comments.